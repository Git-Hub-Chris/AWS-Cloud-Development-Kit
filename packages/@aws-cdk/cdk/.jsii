{
  "author": {
    "name": "Amazon Web Services",
    "organization": true,
    "roles": [
      "author"
    ],
    "url": "https://aws.amazon.com"
  },
  "description": "Deprecated: this module ha been renamed to @aws-cdk/core",
  "docs": {
    "deprecated": "This module has been renamed to @aws-cdk/core",
    "stability": "deprecated"
  },
  "homepage": "https://github.com/awslabs/aws-cdk",
  "jsiiVersion": "0.11.3 (build 464b8c6)",
  "license": "Apache-2.0",
  "name": "@aws-cdk/cdk",
  "readme": {
    "markdown": "This module has been renamed to @aws-cdk/core\n<!--BEGIN STABILITY BANNER-->\n\n---\n\n![Stability: Deprecated](https://img.shields.io/badge/stability-Deprecated-critical.svg?style=for-the-badge)\n\n> This API may emit warnings. Backward compatibility is not guaranteed.\n\n---\n<!--END STABILITY BANNER-->\n\n"
  },
  "repository": {
    "directory": "packages/@aws-cdk/cdk",
    "type": "git",
    "url": "https://github.com/awslabs/aws-cdk.git"
  },
  "schema": "jsii/0.10.0",
  "targets": {
    "java": {
      "maven": {
        "artifactId": "cdk",
        "groupId": "software.amazon.awscdk"
      },
      "package": "software.amazon.awscdk"
    },
    "js": {
      "npm": "@aws-cdk/cdk"
    },
    "python": {
      "distName": "aws-cdk.cdk",
      "module": "aws_cdk.cdk"
    }
  },
  "types": {},
  "version": "0.35.0",
  "fingerprint": "ia0gOKzBkyR6GiGt0pvF9/ZEsI2wa9dGr/PZyezy1zI="
}
