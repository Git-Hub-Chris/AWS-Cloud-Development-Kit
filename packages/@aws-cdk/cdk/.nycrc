/Users/benisrae/code/cdk/aws-cdk-2/tools/cdk-build-tools/config/nycrc