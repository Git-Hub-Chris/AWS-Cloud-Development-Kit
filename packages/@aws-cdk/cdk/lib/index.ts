export * from './aspect';
export * from './tag-aspect';

export * from './construct';
export * from './token';
export * from './unresolved';
export * from './tag-manager';
export * from './dependency';

export * from './cloudformation/cloudformation-json';
export * from './cloudformation/reference';
export * from './cloudformation/cfn-condition';
export * from './cloudformation/fn';
export * from './cloudformation/include';
export * from './cloudformation/logical-id';
export * from './cloudformation/cfn-mapping';
export * from './cloudformation/cfn-output';
export * from './cloudformation/cfn-parameter';
export * from './cloudformation/pseudo';
export * from './cloudformation/cfn-resource';
export * from './cloudformation/resource-policy';
export * from './cloudformation/cfn-rule';
export * from './cloudformation/stack';
export * from './cloudformation/cfn-element';
export * from './cloudformation/dynamic-reference';
export * from './cloudformation/tag';
export * from './cloudformation/removal-policy';
export * from './cloudformation/arn';
export * from './cloudformation/secret';

export * from './app';
export * from './context';
export * from './environment';

export * from './runtime';

export * from './synthesis';
