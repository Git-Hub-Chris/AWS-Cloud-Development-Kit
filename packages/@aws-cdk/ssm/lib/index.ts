// The L1 Library for AWS::SSM:
export * from '../cfn/ssm';
