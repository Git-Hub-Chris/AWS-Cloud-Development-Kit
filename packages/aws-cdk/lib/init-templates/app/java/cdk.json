{
  "app": "mvn compile -q && mvn -e -q exec:java"
}
