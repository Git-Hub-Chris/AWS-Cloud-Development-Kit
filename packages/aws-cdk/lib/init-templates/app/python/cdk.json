{
    "app": "python3 hello/hello_stack.py"
}
