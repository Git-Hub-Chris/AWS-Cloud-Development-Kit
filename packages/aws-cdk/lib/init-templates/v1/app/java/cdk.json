{
  "app": "mvn -e -q compile exec:java",
  "watch": {
    "exclude": [
      "target",
      "src/test",
      "pom.xml",
      "README.md",
      "cdk.json"
    ]
  }
}
