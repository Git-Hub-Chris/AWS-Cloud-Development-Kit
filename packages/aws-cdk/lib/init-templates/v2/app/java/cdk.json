{
  "app": "mvn -e -q compile exec:java",
  "watch": {
    "include": ["**"],
    "exclude": [
      "target",
      "src/test",
      "pom.xml",
      "README.md",
      "cdk.json"
    ]
  }
}
