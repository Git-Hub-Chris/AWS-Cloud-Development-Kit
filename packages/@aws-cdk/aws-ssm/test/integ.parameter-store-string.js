"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const cdk = require("@aws-cdk/core");
const integ = require("@aws-cdk/integ-tests");
const ssm = require("../lib");
const SECURE_PARAM_NAME = '/My/Secret/Parameter';
class CreatingStack extends cdk.Stack {
    constructor(scope, id) {
        super(scope, id);
        new ssm.StringParameter(this, 'String', {
            parameterName: '/My/Public/Parameter',
            stringValue: 'Abc123',
        });
        new integ.AwsApiCall(this, 'SecureParam', {
            service: 'SSM',
            api: 'putParameter',
            parameters: {
                Name: SECURE_PARAM_NAME,
                Type: 'SecureString',
                Value: 'Abc123',
            },
        });
    }
}
class UsingStack extends cdk.Stack {
    constructor(scope, id) {
        super(scope, id);
        // Parameter that contains version number, will be used to pass
        // version value from token.
        const parameterVersion = new cdk.CfnParameter(this, 'MyParameterVersion', {
            type: 'Number',
            default: 1,
        }).valueAsNumber;
        // Retrieve the latest value of the non-secret parameter
        // with name "/My/String/Parameter".
        const stringValue = ssm.StringParameter.fromStringParameterAttributes(this, 'MyValue', {
            parameterName: '/My/Public/Parameter',
        }).stringValue;
        const stringValueVersionFromToken = ssm.StringParameter.fromStringParameterAttributes(this, 'MyValueVersionFromToken', {
            parameterName: '/My/Public/Parameter',
            // parameter version from token
            version: parameterVersion,
        }).stringValue;
        // Retrieve a specific version of the secret (SecureString) parameter.
        const secretValue = ssm.StringParameter.fromSecureStringParameterAttributes(this, 'MySecureValue', {
            parameterName: '/My/Secret/Parameter',
        }).stringValue;
        const secretValueVersion = ssm.StringParameter.fromSecureStringParameterAttributes(this, 'MySecureValueVersion', {
            parameterName: '/My/Secret/Parameter',
            version: 1,
        }).stringValue;
        const secretValueVersionFromToken = ssm.StringParameter.fromSecureStringParameterAttributes(this, 'MySecureValueVersionFromToken', {
            parameterName: '/My/Secret/Parameter',
            // parameter version from token
            version: parameterVersion,
        }).stringValue;
        const user = new cdk.CfnResource(this, 'DummyResourceUsingStringParameters', {
            type: 'AWS::IAM::User',
            properties: {
                LoginProfile: {
                    Password: cdk.Fn.join('-', [
                        stringValue,
                        stringValueVersionFromToken,
                        secretValue,
                        secretValueVersion,
                        secretValueVersionFromToken,
                    ]),
                },
            },
        });
        user.applyRemovalPolicy(cdk.RemovalPolicy.DESTROY);
    }
}
const app = new cdk.App();
const creating = new CreatingStack(app, 'sspms-creating');
const using = new UsingStack(app, 'sspms-using');
using.addDependency(creating);
const cleanup = new cdk.Stack(app, 'sspms-cleanup');
cleanup.addDependency(using);
const integTest = new integ.IntegTest(app, 'SSMParameterStoreTest', {
    assertionStack: cleanup,
    testCases: [using],
});
integTest.assertions.awsApiCall('SSM', 'deleteParameter', {
    Name: SECURE_PARAM_NAME,
});
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW50ZWcucGFyYW1ldGVyLXN0b3JlLXN0cmluZy5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbImludGVnLnBhcmFtZXRlci1zdG9yZS1zdHJpbmcudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7QUFBQSxxQ0FBcUM7QUFDckMsOENBQThDO0FBQzlDLDhCQUE4QjtBQUU5QixNQUFNLGlCQUFpQixHQUFHLHNCQUFzQixDQUFDO0FBRWpELE1BQU0sYUFBYyxTQUFRLEdBQUcsQ0FBQyxLQUFLO0lBQ25DLFlBQVksS0FBYyxFQUFFLEVBQVU7UUFDcEMsS0FBSyxDQUFDLEtBQUssRUFBRSxFQUFFLENBQUMsQ0FBQztRQUVqQixJQUFJLEdBQUcsQ0FBQyxlQUFlLENBQUMsSUFBSSxFQUFFLFFBQVEsRUFBRTtZQUN0QyxhQUFhLEVBQUUsc0JBQXNCO1lBQ3JDLFdBQVcsRUFBRSxRQUFRO1NBQ3RCLENBQUMsQ0FBQztRQUVILElBQUksS0FBSyxDQUFDLFVBQVUsQ0FBQyxJQUFJLEVBQUUsYUFBYSxFQUFFO1lBQ3hDLE9BQU8sRUFBRSxLQUFLO1lBQ2QsR0FBRyxFQUFFLGNBQWM7WUFDbkIsVUFBVSxFQUFFO2dCQUNWLElBQUksRUFBRSxpQkFBaUI7Z0JBQ3ZCLElBQUksRUFBRSxjQUFjO2dCQUNwQixLQUFLLEVBQUUsUUFBUTthQUNoQjtTQUNGLENBQUMsQ0FBQztLQUNKO0NBQ0Y7QUFFRCxNQUFNLFVBQVcsU0FBUSxHQUFHLENBQUMsS0FBSztJQUNoQyxZQUFZLEtBQWMsRUFBRSxFQUFVO1FBQ3BDLEtBQUssQ0FBQyxLQUFLLEVBQUUsRUFBRSxDQUFDLENBQUM7UUFFakIsK0RBQStEO1FBQy9ELDRCQUE0QjtRQUM1QixNQUFNLGdCQUFnQixHQUFHLElBQUksR0FBRyxDQUFDLFlBQVksQ0FBQyxJQUFJLEVBQUUsb0JBQW9CLEVBQUU7WUFDeEUsSUFBSSxFQUFFLFFBQVE7WUFDZCxPQUFPLEVBQUUsQ0FBQztTQUNYLENBQUMsQ0FBQyxhQUFhLENBQUM7UUFFakIsd0RBQXdEO1FBQ3hELG9DQUFvQztRQUNwQyxNQUFNLFdBQVcsR0FBRyxHQUFHLENBQUMsZUFBZSxDQUFDLDZCQUE2QixDQUFDLElBQUksRUFBRSxTQUFTLEVBQUU7WUFDckYsYUFBYSxFQUFFLHNCQUFzQjtTQUV0QyxDQUFDLENBQUMsV0FBVyxDQUFDO1FBQ2YsTUFBTSwyQkFBMkIsR0FBRyxHQUFHLENBQUMsZUFBZSxDQUFDLDZCQUE2QixDQUFDLElBQUksRUFBRSx5QkFBeUIsRUFBRTtZQUNySCxhQUFhLEVBQUUsc0JBQXNCO1lBQ3JDLCtCQUErQjtZQUMvQixPQUFPLEVBQUUsZ0JBQWdCO1NBQzFCLENBQUMsQ0FBQyxXQUFXLENBQUM7UUFFZixzRUFBc0U7UUFDdEUsTUFBTSxXQUFXLEdBQUcsR0FBRyxDQUFDLGVBQWUsQ0FBQyxtQ0FBbUMsQ0FBQyxJQUFJLEVBQUUsZUFBZSxFQUFFO1lBQ2pHLGFBQWEsRUFBRSxzQkFBc0I7U0FDdEMsQ0FBQyxDQUFDLFdBQVcsQ0FBQztRQUNmLE1BQU0sa0JBQWtCLEdBQUcsR0FBRyxDQUFDLGVBQWUsQ0FBQyxtQ0FBbUMsQ0FBQyxJQUFJLEVBQUUsc0JBQXNCLEVBQUU7WUFDL0csYUFBYSxFQUFFLHNCQUFzQjtZQUNyQyxPQUFPLEVBQUUsQ0FBQztTQUNYLENBQUMsQ0FBQyxXQUFXLENBQUM7UUFDZixNQUFNLDJCQUEyQixHQUFHLEdBQUcsQ0FBQyxlQUFlLENBQUMsbUNBQW1DLENBQUMsSUFBSSxFQUFFLCtCQUErQixFQUFFO1lBQ2pJLGFBQWEsRUFBRSxzQkFBc0I7WUFDckMsK0JBQStCO1lBQy9CLE9BQU8sRUFBRSxnQkFBZ0I7U0FDMUIsQ0FBQyxDQUFDLFdBQVcsQ0FBQztRQUVmLE1BQU0sSUFBSSxHQUFHLElBQUksR0FBRyxDQUFDLFdBQVcsQ0FBQyxJQUFJLEVBQUUsb0NBQW9DLEVBQUU7WUFDM0UsSUFBSSxFQUFFLGdCQUFnQjtZQUN0QixVQUFVLEVBQUU7Z0JBQ1YsWUFBWSxFQUFFO29CQUNaLFFBQVEsRUFBRSxHQUFHLENBQUMsRUFBRSxDQUFDLElBQUksQ0FBQyxHQUFHLEVBQUU7d0JBQ3pCLFdBQVc7d0JBQ1gsMkJBQTJCO3dCQUMzQixXQUFXO3dCQUNYLGtCQUFrQjt3QkFDbEIsMkJBQTJCO3FCQUM1QixDQUFDO2lCQUNIO2FBQ0Y7U0FDRixDQUFDLENBQUM7UUFDSCxJQUFJLENBQUMsa0JBQWtCLENBQUMsR0FBRyxDQUFDLGFBQWEsQ0FBQyxPQUFPLENBQUMsQ0FBQztLQUNwRDtDQUNGO0FBRUQsTUFBTSxHQUFHLEdBQUcsSUFBSSxHQUFHLENBQUMsR0FBRyxFQUFFLENBQUM7QUFFMUIsTUFBTSxRQUFRLEdBQUcsSUFBSSxhQUFhLENBQUMsR0FBRyxFQUFFLGdCQUFnQixDQUFDLENBQUM7QUFFMUQsTUFBTSxLQUFLLEdBQUcsSUFBSSxVQUFVLENBQUMsR0FBRyxFQUFFLGFBQWEsQ0FBQyxDQUFDO0FBQ2pELEtBQUssQ0FBQyxhQUFhLENBQUMsUUFBUSxDQUFDLENBQUM7QUFFOUIsTUFBTSxPQUFPLEdBQUcsSUFBSSxHQUFHLENBQUMsS0FBSyxDQUFDLEdBQUcsRUFBRSxlQUFlLENBQUMsQ0FBQztBQUNwRCxPQUFPLENBQUMsYUFBYSxDQUFDLEtBQUssQ0FBQyxDQUFDO0FBRTdCLE1BQU0sU0FBUyxHQUFHLElBQUksS0FBSyxDQUFDLFNBQVMsQ0FBQyxHQUFHLEVBQUUsdUJBQXVCLEVBQUU7SUFDbEUsY0FBYyxFQUFFLE9BQU87SUFDdkIsU0FBUyxFQUFFLENBQUMsS0FBSyxDQUFDO0NBQ25CLENBQUMsQ0FBQztBQUVILFNBQVMsQ0FBQyxVQUFVLENBQUMsVUFBVSxDQUFDLEtBQUssRUFBRSxpQkFBaUIsRUFBRTtJQUN4RCxJQUFJLEVBQUUsaUJBQWlCO0NBQ3hCLENBQUMsQ0FBQyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCAqIGFzIGNkayBmcm9tICdAYXdzLWNkay9jb3JlJztcbmltcG9ydCAqIGFzIGludGVnIGZyb20gJ0Bhd3MtY2RrL2ludGVnLXRlc3RzJztcbmltcG9ydCAqIGFzIHNzbSBmcm9tICcuLi9saWInO1xuXG5jb25zdCBTRUNVUkVfUEFSQU1fTkFNRSA9ICcvTXkvU2VjcmV0L1BhcmFtZXRlcic7XG5cbmNsYXNzIENyZWF0aW5nU3RhY2sgZXh0ZW5kcyBjZGsuU3RhY2sge1xuICBjb25zdHJ1Y3RvcihzY29wZTogY2RrLkFwcCwgaWQ6IHN0cmluZykge1xuICAgIHN1cGVyKHNjb3BlLCBpZCk7XG5cbiAgICBuZXcgc3NtLlN0cmluZ1BhcmFtZXRlcih0aGlzLCAnU3RyaW5nJywge1xuICAgICAgcGFyYW1ldGVyTmFtZTogJy9NeS9QdWJsaWMvUGFyYW1ldGVyJyxcbiAgICAgIHN0cmluZ1ZhbHVlOiAnQWJjMTIzJyxcbiAgICB9KTtcblxuICAgIG5ldyBpbnRlZy5Bd3NBcGlDYWxsKHRoaXMsICdTZWN1cmVQYXJhbScsIHtcbiAgICAgIHNlcnZpY2U6ICdTU00nLFxuICAgICAgYXBpOiAncHV0UGFyYW1ldGVyJyxcbiAgICAgIHBhcmFtZXRlcnM6IHtcbiAgICAgICAgTmFtZTogU0VDVVJFX1BBUkFNX05BTUUsXG4gICAgICAgIFR5cGU6ICdTZWN1cmVTdHJpbmcnLFxuICAgICAgICBWYWx1ZTogJ0FiYzEyMycsXG4gICAgICB9LFxuICAgIH0pO1xuICB9XG59XG5cbmNsYXNzIFVzaW5nU3RhY2sgZXh0ZW5kcyBjZGsuU3RhY2sge1xuICBjb25zdHJ1Y3RvcihzY29wZTogY2RrLkFwcCwgaWQ6IHN0cmluZykge1xuICAgIHN1cGVyKHNjb3BlLCBpZCk7XG5cbiAgICAvLyBQYXJhbWV0ZXIgdGhhdCBjb250YWlucyB2ZXJzaW9uIG51bWJlciwgd2lsbCBiZSB1c2VkIHRvIHBhc3NcbiAgICAvLyB2ZXJzaW9uIHZhbHVlIGZyb20gdG9rZW4uXG4gICAgY29uc3QgcGFyYW1ldGVyVmVyc2lvbiA9IG5ldyBjZGsuQ2ZuUGFyYW1ldGVyKHRoaXMsICdNeVBhcmFtZXRlclZlcnNpb24nLCB7XG4gICAgICB0eXBlOiAnTnVtYmVyJyxcbiAgICAgIGRlZmF1bHQ6IDEsXG4gICAgfSkudmFsdWVBc051bWJlcjtcblxuICAgIC8vIFJldHJpZXZlIHRoZSBsYXRlc3QgdmFsdWUgb2YgdGhlIG5vbi1zZWNyZXQgcGFyYW1ldGVyXG4gICAgLy8gd2l0aCBuYW1lIFwiL015L1N0cmluZy9QYXJhbWV0ZXJcIi5cbiAgICBjb25zdCBzdHJpbmdWYWx1ZSA9IHNzbS5TdHJpbmdQYXJhbWV0ZXIuZnJvbVN0cmluZ1BhcmFtZXRlckF0dHJpYnV0ZXModGhpcywgJ015VmFsdWUnLCB7XG4gICAgICBwYXJhbWV0ZXJOYW1lOiAnL015L1B1YmxpYy9QYXJhbWV0ZXInLFxuICAgICAgLy8gJ3ZlcnNpb24nIGNhbiBiZSBzcGVjaWZpZWQgYnV0IGlzIG9wdGlvbmFsLlxuICAgIH0pLnN0cmluZ1ZhbHVlO1xuICAgIGNvbnN0IHN0cmluZ1ZhbHVlVmVyc2lvbkZyb21Ub2tlbiA9IHNzbS5TdHJpbmdQYXJhbWV0ZXIuZnJvbVN0cmluZ1BhcmFtZXRlckF0dHJpYnV0ZXModGhpcywgJ015VmFsdWVWZXJzaW9uRnJvbVRva2VuJywge1xuICAgICAgcGFyYW1ldGVyTmFtZTogJy9NeS9QdWJsaWMvUGFyYW1ldGVyJyxcbiAgICAgIC8vIHBhcmFtZXRlciB2ZXJzaW9uIGZyb20gdG9rZW5cbiAgICAgIHZlcnNpb246IHBhcmFtZXRlclZlcnNpb24sXG4gICAgfSkuc3RyaW5nVmFsdWU7XG5cbiAgICAvLyBSZXRyaWV2ZSBhIHNwZWNpZmljIHZlcnNpb24gb2YgdGhlIHNlY3JldCAoU2VjdXJlU3RyaW5nKSBwYXJhbWV0ZXIuXG4gICAgY29uc3Qgc2VjcmV0VmFsdWUgPSBzc20uU3RyaW5nUGFyYW1ldGVyLmZyb21TZWN1cmVTdHJpbmdQYXJhbWV0ZXJBdHRyaWJ1dGVzKHRoaXMsICdNeVNlY3VyZVZhbHVlJywge1xuICAgICAgcGFyYW1ldGVyTmFtZTogJy9NeS9TZWNyZXQvUGFyYW1ldGVyJyxcbiAgICB9KS5zdHJpbmdWYWx1ZTtcbiAgICBjb25zdCBzZWNyZXRWYWx1ZVZlcnNpb24gPSBzc20uU3RyaW5nUGFyYW1ldGVyLmZyb21TZWN1cmVTdHJpbmdQYXJhbWV0ZXJBdHRyaWJ1dGVzKHRoaXMsICdNeVNlY3VyZVZhbHVlVmVyc2lvbicsIHtcbiAgICAgIHBhcmFtZXRlck5hbWU6ICcvTXkvU2VjcmV0L1BhcmFtZXRlcicsXG4gICAgICB2ZXJzaW9uOiAxLFxuICAgIH0pLnN0cmluZ1ZhbHVlO1xuICAgIGNvbnN0IHNlY3JldFZhbHVlVmVyc2lvbkZyb21Ub2tlbiA9IHNzbS5TdHJpbmdQYXJhbWV0ZXIuZnJvbVNlY3VyZVN0cmluZ1BhcmFtZXRlckF0dHJpYnV0ZXModGhpcywgJ015U2VjdXJlVmFsdWVWZXJzaW9uRnJvbVRva2VuJywge1xuICAgICAgcGFyYW1ldGVyTmFtZTogJy9NeS9TZWNyZXQvUGFyYW1ldGVyJyxcbiAgICAgIC8vIHBhcmFtZXRlciB2ZXJzaW9uIGZyb20gdG9rZW5cbiAgICAgIHZlcnNpb246IHBhcmFtZXRlclZlcnNpb24sXG4gICAgfSkuc3RyaW5nVmFsdWU7XG5cbiAgICBjb25zdCB1c2VyID0gbmV3IGNkay5DZm5SZXNvdXJjZSh0aGlzLCAnRHVtbXlSZXNvdXJjZVVzaW5nU3RyaW5nUGFyYW1ldGVycycsIHtcbiAgICAgIHR5cGU6ICdBV1M6OklBTTo6VXNlcicsXG4gICAgICBwcm9wZXJ0aWVzOiB7XG4gICAgICAgIExvZ2luUHJvZmlsZToge1xuICAgICAgICAgIFBhc3N3b3JkOiBjZGsuRm4uam9pbignLScsIFtcbiAgICAgICAgICAgIHN0cmluZ1ZhbHVlLFxuICAgICAgICAgICAgc3RyaW5nVmFsdWVWZXJzaW9uRnJvbVRva2VuLFxuICAgICAgICAgICAgc2VjcmV0VmFsdWUsXG4gICAgICAgICAgICBzZWNyZXRWYWx1ZVZlcnNpb24sXG4gICAgICAgICAgICBzZWNyZXRWYWx1ZVZlcnNpb25Gcm9tVG9rZW4sXG4gICAgICAgICAgXSksXG4gICAgICAgIH0sXG4gICAgICB9LFxuICAgIH0pO1xuICAgIHVzZXIuYXBwbHlSZW1vdmFsUG9saWN5KGNkay5SZW1vdmFsUG9saWN5LkRFU1RST1kpO1xuICB9XG59XG5cbmNvbnN0IGFwcCA9IG5ldyBjZGsuQXBwKCk7XG5cbmNvbnN0IGNyZWF0aW5nID0gbmV3IENyZWF0aW5nU3RhY2soYXBwLCAnc3NwbXMtY3JlYXRpbmcnKTtcblxuY29uc3QgdXNpbmcgPSBuZXcgVXNpbmdTdGFjayhhcHAsICdzc3Btcy11c2luZycpO1xudXNpbmcuYWRkRGVwZW5kZW5jeShjcmVhdGluZyk7XG5cbmNvbnN0IGNsZWFudXAgPSBuZXcgY2RrLlN0YWNrKGFwcCwgJ3NzcG1zLWNsZWFudXAnKTtcbmNsZWFudXAuYWRkRGVwZW5kZW5jeSh1c2luZyk7XG5cbmNvbnN0IGludGVnVGVzdCA9IG5ldyBpbnRlZy5JbnRlZ1Rlc3QoYXBwLCAnU1NNUGFyYW1ldGVyU3RvcmVUZXN0Jywge1xuICBhc3NlcnRpb25TdGFjazogY2xlYW51cCxcbiAgdGVzdENhc2VzOiBbdXNpbmddLFxufSk7XG5cbmludGVnVGVzdC5hc3NlcnRpb25zLmF3c0FwaUNhbGwoJ1NTTScsICdkZWxldGVQYXJhbWV0ZXInLCB7XG4gIE5hbWU6IFNFQ1VSRV9QQVJBTV9OQU1FLFxufSk7XG4iXX0=