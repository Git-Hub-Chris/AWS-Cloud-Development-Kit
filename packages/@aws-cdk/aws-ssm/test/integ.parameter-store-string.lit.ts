import cdk = require('@aws-cdk/cdk');
import ssm = require('../lib');

class CreatingStack extends cdk.Stack {
  constructor(scope: cdk.App, id: string) {
    super(scope, id);

    new ssm.StringParameter(this, 'String', {
      name: '/My/String/Parameter',
      stringValue: 'abcdef'
    });
  }
}

class UsingStack extends cdk.Stack {
  constructor(scope: cdk.App, id: string) {
    super(scope, id);

    /// !show
    // Retrieve the value with name "/My/String/Parameter"
    const stringValue = new ssm.ParameterStoreString(this, 'MyValue', {
      parameterName: '/My/String/Parameter'
    }).stringValue;
    /// !hide

    new cdk.Output(this, 'TheValue', { value: stringValue });
  }
}

const app = new cdk.App();

const creating = new CreatingStack(app, 'sspms-creating');
const using = new UsingStack(app, 'sspms-using');
using.addDependency(creating);

app.run();