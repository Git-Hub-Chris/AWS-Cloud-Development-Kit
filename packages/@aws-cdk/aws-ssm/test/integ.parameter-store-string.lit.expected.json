[
  {
    "Resources": {
      "String0BA8456E": {
        "Type": "AWS::SSM::Parameter",
        "Properties": {
          "Type": "String",
          "Value": "abcdef",
          "Name": "/My/String/Parameter"
        }
      }
    }
  },
  {
    "Parameters": {
      "MyValueParameterCCC58B13": {
        "Type": "AWS::SSM::Parameter::Value<String>",
        "Default": "/My/String/Parameter"
      }
    },
    "Outputs": {
      "TheValue": {
        "Value": {
          "Ref": "MyValueParameterCCC58B13"
        }
      }
    }
  }
]